test/extended/openstack/utils.go

```diff
@@ -0,0 +1,108 @@
+package openstack
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"strconv"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	o "github.com/onsi/gomega"
+)
+
+type KuryrNetwork struct {
+	Status struct {
+		SubnetID string `json:"subnetId"`
+	} `json:"status"`
+}
+
+func ElementExists(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func RandomSuffix() string {
+	return strconv.Itoa(rand.Intn(10000))
+}
+
+func GetKuryrNetwork(clientSet *kubernetes.Clientset, namespace string) (KuryrNetwork, error) {
+	//TODO(itzikb): Replace a direct call to the API by extending the ClientSet
+	knetworkPath := fmt.Sprintf("/apis/openstack.org/v1/namespaces/%s/kuryrnetworks/%s", namespace, namespace)
+	data, err := clientSet.RESTClient().
+		Get().
+		AbsPath(knetworkPath).
+		DoRaw(context.TODO())
+	var kn KuryrNetwork
+	json.Unmarshal(data, &kn)
+	return kn, err
+}
+
+func GetSubnetIDfromKuryrNetwork(clientSet *kubernetes.Clientset, namespace string) (string, error) {
+	kn, err := GetKuryrNetwork(clientSet, namespace)
+	if err != nil {
+		return "", err
+	}
+	if (kn != KuryrNetwork{}) {
```

Comment:
there is no need for this check since a Pod has been created Prior to `GetSubnetIDfromKuryrNetwork` there must be a KuryrNetwork CR

Reply:
Done